### PR TITLE
feat(jans-cedarling): Token Validation and Non-Deterministic Detection

### LINKED ISSUE #12113: feat(jans-cedarling): Token Input Validation
**Is your feature request related to a problem? Please describe.**
Follow up #12112 
Implement validation for the token input array, including mapping format validation and JWT payload structure checks.

**Tasks:**
[ ] Create TokenInputValidator struct
[ ] Validate mapping string format (e.g., "Jans::Access_Token")
[ ] Validate JWT payload structure (basic format check)
[ ] Add validation for non-empty mapping strings
[ ] Write unit tests

**Acceptance Criteria:**
[ ] Invalid mapping formats are rejected with clear errors
[ ] Malformed JWT payloads are detected
[ ] Empty or null mappings are rejected
[ ] All edge cases are covered by tests

### PR DESCRIPTION
**Pull request by 0xtinkle**
### Prepare

- [x] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [x] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

### Description
This PR implements the token validation logic and non-deterministic token detection for the `authorize_multi_issuer` feature. It builds upon the core data structures from #12130  to provide comprehensive JWT parsing, validation, and duplicate token detection capabilities.

#### Target issue
  
  
  
  
closes #12113 , #12118 

#### Implementation Details
  

**Token Validation Logic:**
- JWT parsing and payload extraction using existing `decode_jwt` function
- Token type extraction using full mapping string (e.g., "Jans::Access_Token" → "Jans::Access_Token")
- Issuer claim validation and extraction from JWT payload
- Integration with existing JWT validation infrastructure

**Non-Deterministic Token Detection:**
- Prevents duplicate token types from the same issuer
- Comprehensive validation that continues processing even with invalid tokens
- Logging integration for failed token validation

-------------------
### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [x] Relevant unit and integration tests have been added/updated
- [x] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [x] **I confirm that there is no impact on the docs due to the code changes in this PR.**

### CODE REVIEW

**Comment by mo-auto** [comment]
### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/janssenproject/pr-checks/d2d76c66-7002-4065-bdc0-99d8e0214dce?refs=94e3f82c5db4e244d8a599b9be685d5fbd9c9cfa&source=prChecksComment)

File: jans-cedarling/cedarling/src/authz/request.rs [hunk 1]
```diff
@@ -199,16 +194,59 @@ impl AuthorizeMultiIssuerRequest {
...
+                        index,
+                        error: err.to_string(),
+                    });
+                },
+            }
+        }
+
+        // If no tokens were successfully validated, return an error
+        if validated_tokens.is_empty() {
+            return Err(MultiIssuerValidationError::EmptyTokenArray);
```
Comment (olehbozhok) [inline jans-cedarling/cedarling/src/authz/request.rs]: I think at this point we need return better error message. Something like: "Could not validate tokens." and maybe return list of tokens type.

And maybe we need log info with full error about token before this check.

File: jans-cedarling/cedarling/src/authz/request.rs [hunk 2]
```diff
@@ -123,39 +126,32 @@ pub struct TokenInput {
...
     /// Create a new TokenInput
     pub fn new(mapping: String, payload: String) -> Self {
         Self { mapping, payload }
     }
 
-    /// Validate the token input
-    pub fn validate(&self) -> Result<(), TokenInputError> {
-        // Check if mapping is empty
+    /// Parse and validate the token, returning (token_type, parsed_payload)
+    pub fn parse_and_validate(&self) -> Result<(String, Value), TokenInputError> {
```
Comment (olehbozhok) [inline jans-cedarling/cedarling/src/authz/request.rs:136]: I think we can handle parsing and validation in the `new` method. We could also add a custom serde deserializer to allow deserialization from JSON, but that's up to you.

**Comment by 0xtinkle** [comment]
> i still don't understand how this closes this issue #12114 if its not a typo could you please explain

Oh damn, yeah I meant #12118
